pymatgen/core/structure.py

```diff
@@ -764,6 +764,28 @@ def get_neighbors(self, site, r, include_index=False):
                                       include_index=include_index)
         return [d for d in nn if site != d[0]]
 
+    def get_nearest_site(self, coords, site, r = None):
```

Comment:
I've also found this function useful for getting the minimum distance between sites of two similar structures (with identically ordered sites) that do not have identical lattices (example script below). Should I instead create a PolarizationLattice object that inherits from Structure and add this method?

```python
def get_min_distances(struct1, struct2):
    assert len(struct1) == len(struct2)
    L = len(struct1)
    dists = []
    for i in range(L):
        coord1 = struct1[i].coords
        coord2 = struct2.get_nearest_site(coord1, struct2[i])[0].coords
        dists.append(np.linalg.norm(coord1-coord2))
    return dists
```